### PR TITLE
opt to disable build record upload

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/actions-toolkit/pull/394
fixes https://github.com/docker/build-push-action/issues/1151

Adds `DOCKER_BUILD_RECORD_UPLOAD` environment variable to enable/disable build record upload as GitHub artifact (defaults to `true`).

Also rename `DOCKER_BUILD_EXPORT_RETENTION_DAYS` to `DOCKER_BUILD_RECORD_RETENTION_DAYS` to be consistent. We keep and mark as deprecated the old one.

Example: https://github.com/docker/build-push-action/actions/runs/9763012530?pr=1172#summary-26947857385

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,7 +259,8 @@ The following outputs are available:
...
 | `DOCKER_BUILD_SUMMARY_DISABLE`       | Bool   | If `true`, [build summary](https://docs.docker.com/build/ci/github-actions/build-summary/) generation is disabled                                                                                                                                                  |
-| `DOCKER_BUILD_EXPORT_RETENTION_DAYS` | Number | Duration after which build export artifact will expire in days. Defaults to repository/org [retention settings](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#artifact-and-log-retention-policy) if unset or `0` |
+| `DOCKER_BUILD_RECORD_DISABLE`        | Bool   | If `true`, build record upload as [GitHub artifact](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts) is disabled.                                                                                                            |
```
Comment (crazy-max) [inline README.md:262]: @colinhemmings @dvdksn I wonder if we should name it `DOCKER_BUILD_RECORD_UPLOAD_DISABLE` to make it more explicit?

Comment (dvdksn) [reply README.md:262]: What if we use `DOCKER_BUILD_RECORD_UPLOAD`, and default to `true`?

Comment (crazy-max) [reply README.md]: Yes good point, I change it to `DOCKER_BUILD_RECORD_UPLOAD` and have done the same for `DOCKER_BUILD_SUMMARY` (will open another PR for this one)

**Review by dvdksn — APPROVED** [review]
LGTM

**Pull request merged**